src/Makefile.am

```diff
@@ -125,8 +125,9 @@ BITCOIN_CORE_H = \
   consensus/consensus.h \
   consensus/tx_check.h \
   consensus/tx_verify.h \
-  core_io.h \
   core_memusage.h \
+  core_read.h \
+  core_write.h \
```

Comment:
Why don't you keep the header and make this core_io.cpp?

Reply:
That is an option but it would introduce 18 (9 * 2) additional unused includes. Is it worth that cost?

Reasoning:

`core_write.h` includes `string`, `amount.h` and `stdint.h` (via `amount.h`).

`core_read.h` includes `string`, `vector` and `attributes.h`.

The following files include only one of `core_read.h` and `core_write.h`:

```
$ git grep -E '^#include <(core_read|core_write)\.h>' -- "*.cpp" "*.h" | \
    cut -f1 -d: | uniq -c | grep ' 1 '
      1 src/core_read.cpp
      1 src/core_write.cpp
      1 src/qt/transactiontablemodel.cpp
      1 src/rpc/blockchain.cpp
      1 src/rpc/net.cpp
      1 src/test/blockfilter_tests.cpp
      1 src/test/rpc_tests.cpp
      1 src/test/transaction_tests.cpp
      1 src/wallet/rpcdump.cpp
```

Reply:
These seem like 18 very small includes. I'm fine with either splitting in `read`/`write` or having one `io`.

Reply:
Agree that they are not worth the effort, also you can avoid them by putting `using CAmount=...` instead of the full header.

Reply:
I don't feel strongly about this. I'll let others chime in and update according to the consensus opinion.

@Sjors @MarcoFalke I take it I have your Concept ACKs? Mind making them explicit in standalone comments?